### ISSUE TITLE
Issue Reproducing RestrictedImageNet Results from Table 3

### ISSUE THREAD

**Issue by xuwangyin** [M4xim4l/InNOutRobustness#4]
Hi,

Thank you for making the code for your paper publicly available! I really appreciated the clarity and quality of your code.

I have been trying to reproduce the RestrictedImageNet results for the AT_3.5 approach reported in Table 3 of the paper. However, I'm encountering a significant discrepancy with the clean accuracy.

**Discrepancy Details**
According to Table 3, the expected results are:
- Clean Accuracy: 93.49%
- L2-3.5 Robust Accuracy: 47.66%

The best results I was able to achieve are:
- Clean Accuracy: ~83%
- L2-3.5 Robust Accuracy: 49.78% (APGD-CE)

While the robust accuracy is in the expected range, the clean accuracy is over 10 percentage points lower than reported.

**Steps to Reproduce**

I followed the hyperparameter settings outlined in Appendix B.4.

1. LR Schedule Modification:
To align with the training details in Appendix B.4, I modified the learning rate schedule logic in utils/run_file_helpers.py. Specifically, I adjusted the schedule to decay the initial learning rate of 0.1 by a factor of 10 at epochs 30, 60, and 75.

2. Script Used (`run_training_restrictedimagenet.py`):
```
import matplotlib as mpl
mpl.use('Agg')

import os
import torch
import torch.nn as nn

from utils.model_normalization import RestrictedImageNetWrapper
import utils.datasets as dl
import utils.models.model_factory_224 as factory
import utils.run_file_helpers as rh
from distutils.util import strtobool

import argparse

parser = argparse.ArgumentParser(description='Define hyperparameters.', prefix_chars='-')
parser.add_argument('--net', type=str, default='resnet50_timm', help='ResNet18, 34, 50 or 101')
parser.add_argument('--model_params', nargs='+', default=[])
parser.add_argument('--dataset', type=str, default='restrictedimagenet', help='restrictedimagenet')
parser.add_argument('--od_dataset', type=str, default='restrictedimagenetOD',
                    help=('restrictedimagenetOD, imagenet or openImages'))
parser.add_argument('--balanced_sampling', type=lambda x: bool(strtobool(x)), default=True,
                    help='Whether to use balanced sampling for dataset')
parser.add_argument('--task', type=str, default='RestrictedImageNet',
                    help='Task name used for model and log directory naming')

rh.parser_add_commons(parser)
rh.parser_add_adversarial_commons(parser)
rh.parser_add_adversarial_norms(parser, 'restrictedimagenet')

hps = parser.parse_args()
#
device_ids = None
if len(hps.gpu)==0:
    device = torch.device('cpu')
    print('Warning! Computing on CPU')
elif len(hps.gpu)==1:
    device = torch.device('cuda:' + str(hps.gpu[0]))
else:
    device_ids = [int(i) for i in hps.gpu]
    device = torch.device('cuda:' + str(min(device_ids)))

#Load model
model_root_dir = f'{hps.task}Models'
logs_root_dir = f'{hps.task}Logs'
num_classes = 9  # RestrictedImageNet typically has 9 classes

if len(hps.model_params) == 0:
    model_params = None
else:
    model_params = hps.model_params
print(model_params)
model, model_name, model_config = factory.build_model(hps.net, num_classes, model_params=model_params)
img_size = 224
model_dir = os.path.join(model_root_dir, model_name)
log_dir = os.path.join(logs_root_dir, model_name)

start_epoch, optim_state_dict = rh.load_model_checkpoint(model, model_dir, device, hps)
model = RestrictedImageNetWrapper(model).to(device)

msda_config = rh.create_msda_config(hps)

#load dataset
od_bs = int(hps.od_bs_factor * hps.bs)

id_config = {}
if hps.dataset == 'restrictedImagenet':
    train_loader = dl.get_restrictedImageNet(train=True, batch_size=hps.bs, augm_type=hps.augm, size=img_size,
                                     config_dict=id_config, balanced=hps.balanced_sampling)
else:
    raise ValueError(f'Dataset {hps.dataset} not supported')

if hps.train_type.lower() in ['ceda', 'acet', 'advacet', 'tradesacet', 'tradesceda']:
    od_config = {}
    loader_config = {'ID config': id_config, 'OD config': od_config}

    if hps.od_dataset == 'restrictedimagenetOD':
        tiny_train = dl.get_restrictedImageNetOD(batch_size=od_bs, augm_type=hps.augm, size=img_size,
                                                 config_dict=od_config)
    elif hps.od_dataset == 'imagenet':
        tiny_train = dl.get_ImageNet(train=True, batch_size=od_bs, augm_type=hps.augm, size=img_size,
                                     exclude_restricted=True, config_dict=od_config)
    elif hps.od_dataset == 'openImages':
        tiny_train = dl.get_openImages('train', batch_size=od_bs, shuffle=True, augm_type=hps.augm,
                                       size=img_size, exclude_dataset='restrictedimagenet', config_dict=od_config)
    else:
        raise ValueError(f'OD dataset {hps.od_dataset} not supported')
else:
    loader_config = {'ID config': id_config}

if hps.dataset == 'restrictedImagenet':
    test_loader = dl.get_restrictedImageNet(train=False, batch_size=hps.bs, augm_type='test', size=img_size, balanced=hps.balanced_sampling)
else:
    assert False

scheduler_config, optimizer_config = rh.create_optim_scheduler_swa_configs(hps)
id_attack_config, od_attack_config = rh.create_attack_config(hps, 'restrictedImagenet')
trainer = rh.create_trainer(hps, model, optimizer_config, scheduler_config, device, num_classes,
                            model_dir, log_dir, msda_config=msda_config, model_config=model_config,
                            id_attack_config=id_attack_config, od_attack_config=od_attack_config)
# import ipdb; ipdb.set_trace()
##DEBUG:
# torch.autograd.set_detect_anomaly(True)
torch.backends.cudnn.benchmark = True

# run training
if trainer.requires_out_distribution():
    train_loaders, test_loaders = trainer.create_loaders_dict(train_loader, test_loader=test_loader,
                                                              out_distribution_loader=tiny_train)
    trainer.train(train_loaders, test_loaders, loader_config=loader_config, start_epoch=start_epoch,
                  optim_state_dict=optim_state_dict, device_ids=device_ids)
else:
    train_loaders, test_loaders = trainer.create_loaders_dict(train_loader, test_loader=test_loader)
    trainer.train(train_loaders, test_loaders, loader_config=loader_config, start_epoch=start_epoch,
                  optim_state_dict=optim_state_dict, device_ids=device_ids)
```
3. Command Used:
```
python -u run_training_restrictedimagenet.py \
  --gpu 0 \
  --net resnet50 \
  --augm madry \
  --stepsize 0.7 \
  --epochs 110 \
  --ema False \
  --ema_decay 0.999 \
  --test_epochs 5 \
  --dataset restrictedImagenet \
  --schedule step_lr \
  --eps 3.5 \
  --od_eps_factor 2 \
  --train_type adversarial \
  --train_clean True \
  --id_steps 7 \
  --id_steps 15 \
  --task RestrictedImageNetMixedTraining
```

**Training Logs**
Here is a screenshot from TensorBoard showing the Test/Statistics/CleanAccuracy and Test/Statistics/IDAccuracy over the training epochs. The clean accuracy plateaus around 83-84%.

<img width="550" alt="Image" src="https://github.com/user-attachments/assets/a6c869ca-e853-4835-b644-a821ea08e864" />

I have double-checked the hyperparameters and my setup, but I can't seem to find the reason for this gap in performance. Could you please advise if I have missed a specific setting or if there is something else that might be causing this discrepancy?

Any help would be greatly appreciated. Thank you!
